### PR TITLE
Add mypy dependency to build requirements in meta.yaml

### PR DESCRIPTION
**Pull request by alopezrivera**

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -3,6 +3,9 @@
 echo "OS is:"
 echo "`uname`"
 
+# Tudat dependencies
+conda install numpy scipy matplotlib tqdm -y
```
Comment (DominicDirkx) [inline recipe/build.sh:7]: Hi @alopezrivera I don't think this should be added here, but rather in the recipe.yamp, to ensure consistency

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {% set name = "tudatpy" %}
   {% set version = "0.8.0.dev9" %}
   {% set build = "0" %}
-  {% set git_rev = "v0.8.0.dev9" %}
+  {% set git_rev = "c51a1b5490df82bec17c3a12108c3f304412a539" %}
```
Comment (DominicDirkx) [inline recipe/meta.yaml:4]: Why this specific commit?

File: recipe/build.sh (hunk 1 above)
Comment (DominicDirkx) [reply recipe/build.sh]: Great to hear from you! Where did you push the changes?

Comment (DominicDirkx) [reply recipe/build.sh]: Great to hear from you! And thanks :)

File: recipe/meta.yaml (hunk 2 above)
Comment (DominicDirkx) [reply recipe/meta.yaml]: I had a look and this was a very old version, which may explain some of the compilation errors. I've updated it

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {% set name = "tudatpy" %}
   {% set version = "0.8.0.dev9" %}
   {% set build = "0" %}
-  {% set git_rev = "v0.8.0.dev9" %}
+  {% set git_rev = "0.8.0.dev9" %}
```
Comment (geoffreygarrett) [inline recipe/meta.yaml:4]: Looks to be an incorrect tag?

**Review by geoffreygarrett — CHANGES_REQUESTED** [review]

**Review by geoffreygarrett — APPROVED** [review]

**Comment by DominicDirkx** [comment]
Hi all, this issue started with the release of v0.8.0.dev9, which caused a whole bunch of issues in the tudatpy build that seem difficult to resolve. I've rolled back the release, and have re-opened the pull request:

https://github.com/tudat-team/tudatpy/pull/154

before merging this, I propose we fix the Azure build of the associated changes, by hard-coding the commit in the recipe.yaml, rather than assigning a version. That way, we can merge other tudatpy features, while we resolve this one on the side. I'll patch up the develop branch of the feedstock.

**Comment by geoffreygarrett** [comment]
AFAK https://github.com/sizmailov/pybind11-stubgen would be the way to go if mypy was intended for stub generation for code completion etc. I've used it successfully in the past.

**Comment by geoffreygarrett** [comment]
I'll close this for now

**Pull request closed without merging**